amethyst_renderer/src/mesh.rs

```diff
@@ -0,0 +1,142 @@
+//! Mesh resource.
+
+use cgmath::{Deg, Matrix4, Point3, Transform, Vector3};
+use error::Result;
+use gfx::Primitive;
+use types::{Factory, RawBuffer, Slice};
+use vertex::{Attribute, VertexFormat};
+
+/// Represents a polygonal mesh.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Mesh {
+    attrs: Vec<Attribute>,
+    prim: Primitive,
+    slice: Slice,
+    transform: Matrix4<f32>,
+    vbuf: RawBuffer,
+}
+
+impl Mesh {
+    /// Builds a new mesh from the given vertices.
+    pub fn new<'v, V: VertexFormat + 'v>(verts: &'v [V]) -> MeshBuilder {
```

Comment:
Well, the `Mesh` needs a constructor of some kind due to the complexity involved. Maybe I could rename it to `from_verts`?